seqr/views/apis/family_api.py

```diff
@@ -43,7 +43,20 @@ def family_page_data(request, family_guid):
     has_case_review_perm = has_case_review_permissions(project, request.user)
 
     sample_models = Sample.objects.filter(individual__family=family)
-    samples = get_json_for_samples(sample_models, project_guid=project.guid, family_guid=family_guid, skip_nested=True, is_analyst=is_analyst)
+    additional_values = {
+        'rnaSeqTpm': Case(When(Exists(RnaSeqTpm.objects.filter(sample_id=OuterRef('pk'))), then=Value('TPM')), default=None),
```

Comment:
So you should be able to do this using the related field on the sample model without having to write query objects with outer refs - something like `When(rnaseqtpm__isnull=False)`

Reply:
I originally had the `__isnull` syntax but the query was timing out with more than 1 rnaseq type. I could add indexes on the foreign keys to the rnaseq tables on the sample model?

Reply:
There should be indices on the foreign keys already, so that sounds like a case where what django is doing is not what we want. You could try using the debug toolbar to see the actual executed SQL and figure out why its so inefficient and adjust it, but honestly that feels like overkill and its probably better not to try to use those fields in this case

That said, if you are going to make 3 distinct queries to each of the subtypes and then iterate through all the sample models and add annotations to them, using the `additional_values` does not really save us any compute and just makes the whole thing less readable in the end. In that case, you are better off with getting lists of sample Ids for each type and then adding to `rnaSeqTypes` based on that, something like

```
rna_type_samples = {
    'TPM': set(RnaSeqTpm.objects.filter(sample__in= sample_models).values_list('sample', flat=True),
    'Expression Outlier': set(RnaSeqOutlier.objects.filter(...
    'Splice Outlier': set(RnaSeqSpliceOutlier.objects.filter(...
}
...
sample['rnaSeqTypes'] = [type for type, sample_ids in rna_type_samples.items() if sample['id'] in sample_ids]
```

Reply:
I implemented it this way ^ which does not require changes to the JS code as you suggested [here](https://github.com/broadinstitute/seqr/pull/4171#discussion_r1651231878). 

Is there a general rule for what belongs in additional_values and when it's better to query related tables separately?